Src/Base/AMReX_PODVector.H

```diff
@@ -212,7 +212,7 @@ namespace amrex
             a_vector.m_capacity = 0;
         }
 
-        ~PODVector () noexcept { deallocate(m_data, capacity()); }
+        ~PODVector () noexcept { if (m_data != nullptr) { deallocate(m_data, capacity()); }}
```

Comment:
For our own allocators, we could also perform this check `if(!nullptr)` in `deallocate`.

That way, `deallocate` would match the behavior of `delete` in std C++: https://en.cppreference.com/w/cpp/language/delete